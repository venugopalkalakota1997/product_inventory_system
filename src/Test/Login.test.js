import React from 'react';
import { MemoryRouter as Router, withRouter } from 'react-router-dom'
import {mount} from 'enzyme';
import {shallow} from 'enzyme';
import Login from '../compoents/Login';
import {configure} from 'enzyme'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter:new Adapter()});

it('Render',()=>{
  const wrapper=shallow(<Login />);
  const h=wrapper.find("h3");
  const result=h.text();

  expect(result).toBe("login Component");
})

test('Login', () => {
    expect(true).toBeTruthy();
  });

it('Submit Fucntion',()=>{
  const onSubmitFun=jest.fn();
  const wrapper=mount(<form onSubmit={onSubmitFun}></form>)
  const form=wrapper.find('form');
  form.simulate('submit');
  expect(onSubmitFun).toHaveBeenCalledTimes(1);
});
it('username ',()=>{
  const wrapper=mount(<input type="text" name="username"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('text');
  expect(input.prop('name')).toEqual('username');
})
it('Password ',()=>{
  const wrapper=mount(<input type="password" name="password"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('password');
  expect(input.prop('name')).toEqual('password');
})
it('Button ',()=>{
  const wrapper=mount(<button type="submit">Login</button>);
  const input=wrapper.find('button');
  const result=input.text();
  expect(result).toBe('Login')
  expect(input.prop('type')).toEqual('submit');
})
test('mathes Snapshot',()=>{
  const tree=renderer.create(<Router><Login /></Router>).toJSON()
  expect(tree).toMatchSnapshot();
})