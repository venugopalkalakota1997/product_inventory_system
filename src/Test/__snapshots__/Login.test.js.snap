// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes Snapshot 1`] = `
<form
  className="form"
>
  <h3>
    login Component
  </h3>
  <p>
    Name 
  </p>
  <input
    className="input"
    id="username"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  <p
    className="error"
  >
    
  </p>
  <p>
    Password 
  </p>
  <input
    className="input"
    id="password"
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
  />
  <br />
  <p
    className="error"
  >
    
  </p>
  <br />
  <button
    className="button"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Login
  </button>
</form>
`;
